arrow/src/array/builder/generic_string_builder.rs

```diff
@@ -28,16 +28,17 @@ pub struct GenericStringBuilder<OffsetSize: OffsetSizeTrait> {
 }
 
 impl<OffsetSize: OffsetSizeTrait> GenericStringBuilder<OffsetSize> {
-    /// Creates a new [`GenericStringBuilder`],
+    /// Creates a new [`GenericStringBuilder`].
     pub fn new() -> Self {
         Self {
             builder: GenericBinaryBuilder::new(),
         }
     }
 
-    /// Creates a new [`GenericStringBuilder`],
-    /// `data_capacity` is the number of bytes of string data to pre-allocate space for in this builder
-    /// `item_capacity` is the number of items to pre-allocate space for in this builder
+    /// Creates a new [`GenericStringBuilder`].
+    ///
+    /// - `item_capacity` is the number of items to pre-allocate (the size of the buffer of offsets).
```

Comment:
```suggestion
    /// - `item_capacity` is the number of items to pre-allocate (i.e. the final length of the array).
```
Or something, the offsets is actually one larger than this 😅

Reply:
Very good point, thanks a lot for the feedback. I added a note about the size of the buffer being one larger, I personally think this should make things very clear, but let me know if you have any other comment.